api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -211,30 +213,92 @@ private void updateUserAccessTiers(DbUser dbUser, Agent agent) {
     tiersForRemoval.forEach(tier -> accessTierService.removeUserFromTier(dbUser, tier));
   }
 
-  public boolean isCompleteAndNotExpired(Timestamp completionTime) {
-    Timestamp expirationTime =
-        new Timestamp(
-            clock.millis()
-                - TimeUnit.MILLISECONDS.convert(
-                    configProvider.get().accessRenewal.expiryDays, TimeUnit.DAYS));
-    if (configProvider.get().access.enableAccessRenewal) {
-      return completionTime != null && expirationTime.before(completionTime);
+  private class ModuleTimes {
```

Comment:
I like this.  Probably makes sense to break it out into its own file.   Maybe under the access package (where I hope all this code will live eventually)

Reply:
I'll merge this and schedule time for a refactoring discussion.